complex objects/cargo/list/src/lib.rs

```diff
@@ -157,74 +172,77 @@ impl ListManager {
         item_list
     }
 
-    pub fn create_item(&self, item: &Item, category_id: i64) -> Option<Item> {
-        println!("Creating item {:?} in category {:?}", item, category_id);
-        let sql = r#"INSERT INTO items (description, created_at, due_date, is_complete, category) VALUES (?, ?, ?, ?, ?)"#;
-        let conn = self.store.write_connection();
+    pub fn fetch_item(&self, uuid: &String) -> Option<Item> {
+        let sql = r#"SELECT uuid, name, due_date, completion_date FROM items WHERE uuid=?"#;
+
+        let conn = self.store.read_connection();
         let mut stmt = conn.prepare(sql).unwrap();
-        match stmt.insert(&[&item.description, &item.created_at, &item.due_date, &item.is_complete, &category_id]) {
-            Ok(row_id) => {
-                let fetch_sql = r#"SELECT id, description, created_at, due_date, is_complete FROM items WHERE rowid=?"#;
-                stmt = conn.prepare(fetch_sql).unwrap();
-                let mut item_iter = stmt.query_map(&[&row_id], |row| {
-                    let complete: i64 = row.get(4);
-                    Item {
-                        id: row.get(0),
-                        description: row.get(1),
-                        created_at: row.get(2),
-                        due_date: row.get(3),
-                        is_complete: complete != 0
-                    }
-                }).unwrap();
-
-                if let Some(result) = item_iter.next() {
-                    result.ok()
-                } else {
-                    println!("No item found");
-                    None
-                }
-            },
-            Err(e) => {
-                println!("Failed to create item {:?}", e);
-                None
+        let mut item_iter = stmt.query_map(&[uuid], |row| {
+            Item {
+                uuid: row.get(0),
+                name: row.get(1),
+                due_date: row.get(2),
+                completion_date: row.get(3),
+                labels: self.fetch_labels_for_item(uuid)
             }
+        }).unwrap();
+
+        if let Some(result) = item_iter.next() {
+            result.ok()
+        } else {
+            println!("No item found for uuid {:?}", uuid);
+            None
         }
     }
 
+    pub fn create_item(&self, item: &Item) -> Option<Item> {
+        println!("Creating item {:?}", item);
+        let item_sql = r#"INSERT INTO items (uuid, name, due_date, completion_date) VALUES (?, ?, ?, ?)"#;
+        let conn = self.store.write_connection();
+        let item_uuid = Uuid::new_v4().simple().to_string();
+        conn.execute(item_sql, &[&item_uuid, &item.name, &item.due_date, &item.completion_date]).unwrap();
+
+        let item_label_sql = r#"INSERT INTO item_labels (item_uuid, label_name) VALUES (?, ?)"#;
```

Comment:
So at the moment we're requiring that an item must have a label, even though the schema doesn't dictate it. I like that.